frontend/src/app/main/ui/ds/tooltip/tooltip.cljs

```diff
@@ -0,0 +1,207 @@
+       :bottom (+ (- trigger-bottom offset) tooltip-height)
+       :width tooltip-width
+       :height tooltip-height}
+
+      "top-right"
+      {:top (- trigger-top offset tooltip-height)
+       :left (- trigger-right overlay-offset)
+       :right (+ (- trigger-right overlay-offset) tooltip-width)
+       :bottom (+ (- trigger-top offset tooltip-height) tooltip-height)
+       :width tooltip-width
+       :height tooltip-height}
+
+      "top-left"
+      {:top (- trigger-top offset tooltip-height)
+       :left (+ (- trigger-left tooltip-width) overlay-offset)
+       :right (+ (- trigger-left tooltip-width) overlay-offset tooltip-width)
+       :bottom (+ (- trigger-top offset tooltip-height) tooltip-height)
+       :width tooltip-width
+       :height tooltip-height}
+
+      {:top (- trigger-top offset tooltip-height)
+       :left (- (+ trigger-left (/ trigger-width 2)) (/ tooltip-width 2))
+       :right (+ (- (+ trigger-left (/ trigger-width 2)) (/ tooltip-width 2)) tooltip-width)
+       :bottom (+ (- trigger-top offset tooltip-height) tooltip-height)
+       :width tooltip-width
+       :height tooltip-height})))
+
+(defn- get-fallback-order [placement]
+  (case placement
+    "top" ["top" "right" "bottom" "left" "top-right" "bottom-right" "bottom-left" "top-left"]
+    "bottom" ["bottom" "left" "top" "right" "bottom-right" "bottom-left" "top-left" "top-right"]
+    "left" ["left" "top" "right" "bottom" "top-left" "top-right" "bottom-right" "bottom-left"]
+    "right" ["right" "bottom" "left" "top" "bottom-left" "top-left" "top-right" "bottom-right"]
+    "top-right" ["top-right" "right" "bottom" "left" "top" "bottom-right" "bottom-left" "top-left"]
+    "bottom-right" ["bottom-right" "bottom" "left" "top" "right" "bottom-left" "top-left" "top-right"]
+    "bottom-left" ["bottom-left" "left" "top" "right" "bottom"  "top-left" "top-right" "bottom-right"]
+    "top-left" ["top-left" "top" "right" "bottom" "left" "bottom-left" "top-right" "bottom-right"]))
+
+(def ^:private schema:tooltip
+  [:map
+   [:class {:optional true} :string]
+   [:id :string]
+   [:offset {:optional true} :int]
+   [:delay {:optional true} :int]
+   [:placement {:optional true}
+    [:maybe [:enum "top" "bottom" "left" "right" "top-right" "bottom-right" "bottom-left" "top-left"]]]])
+
+(mf/defc tooltip*
+  {::mf/props :obj
+   ::mf/schema schema:tooltip}
+  [{:keys [class id children content placement offset delay] :rest props}]
+  (let [placement* (mf/use-state (d/nilv placement "top"))
+        placement  (deref placement*)
+        delay      (d/nilv delay 300)
+
+        schedule-ref (mf/use-ref nil)
+
+        position-tooltip
+        (fn [^js tooltip trigger]
+          (let [all-placements (get-fallback-order placement)]
+            (.showPopover tooltip)
+            (loop [[current-placement & remaining-placements] all-placements]
+              (when current-placement
+                (reset! placement* (name current-placement))
```

Comment:
I think this `name` is not necessary. because current-placement should be already a string.